addons/event/models/event_mail.py

```diff
@@ -44,16 +48,27 @@ class EventTypeMail(models.Model):
         ('before_event', 'Before the event'),
         ('after_event', 'After the event')],
         string='Trigger', default="before_event", required=True)
-    template_id = fields.Many2one(
-        'mail.template', string='Email Template',
-        domain=[('model', '=', 'event.registration')], ondelete='restrict',
-        help='This field contains the template of the mail that will be automatically sent')
+    template_model_id = fields.Many2one('ir.model', string='Template Model', compute='_compute_template_model_id')
+    template_ref = fields.Reference(string='Template', selection='_selection_template_model', required=True, ondelete='cascade')
```

Comment:
How is the ondelete working on a ref ? If it works as a m2O I would keep a restrict to avoid people unlinking the template without knowing it is used in some event communication.

Reply:
Just tested and it does not work... The cascade was there because on the old template_id field, so I remove it